### ISSUE TITLE
Android app loses connection to API

### ISSUE THREAD

**Issue by cjnaz** [xBrowserSync/App#61]
I recently changed my API server from http to https (added LetEncrypt cert), and changed my ddns provider.  Since then my Android connection to the API server fails after several hours ("Status Error") and will not recover by itself.  If I reenter the API service URL, the Android app reports "Not a valid xBrowserSync service".  Note that I can set it to https://api.xbrowsersync.org successfully, although I have not tested a sync.
To recover, I need to clear the app storage/data.  The app will then accept my API service URL, but will fail again after several minutes or hours, seemingly at random.
The API service (v1.1.4) was not changed with the ddns/https change, and worked solid on Android when not running with SSL/TLS.  The API server with SSL/TLS and the new ddns provider continue to work fine/solid for several Chrome and Firefox PC clients from within my LAN or from the WAN.

### Steps to reproduce
1.  Not sure.  I can send you my service URL if that will help.
2.
3.

### Expected behaviour
Tell us what should happen

### Actual behaviour
above

### Client app information
**App Platform:** (Chrome, Firefox, iOS, etc) 
Android 6.0.1

**App Platform Version:** 
v1.4.0

#### Other info
Personal API service running v1.1.4.

**Comment by nero120** [comment]
Odd one, not doing much in the xBrowserSync code besides checking `/info` from your service. Assuming the call returns the service info and the SSL cert is valid it should work fine. Can you browse to your service in Chrome on Android (over HTTPS) when you start getting the error in xBrowserSync?

**Comment by cjnaz** [comment]
Yes, I can browse to the service in Chrome on Android with HTTPS while the Android app says it cannot talk to the service.  This works while on my LAN wifi (local to the server) and over the carrier (Verizon) WAN.  A Windows notebook connected to the phone via Wifi hotspot also connects to the service without error (and can update bookmarks back to the server).
I'll email my service URL for your testing and poking.  Perhaps you can reproduce this from your Android device.

**Comment by nero120** [comment]
After replicating the issue with the service url you emailed me, I discovered that the connection error that gets thrown indicates that the request was aborted for some reason, but give no other information as to why it was aborted. It definitely is not timing out, and the service is online as I can browse to it within chrome and access both the home page and the API routes fine.

I also found that when your service is erroring, I can get it back online by doing the following:

1. Change the service to https://api.xbrowsersync.org
2. Close xBrowserSync completely (you have to swipe it right on the recent apps menu so it's not running in the background).
3. Re-open xBrowserSync.
4. Change the service back to yours and it will connect fine.

However, this will only work until you close the app. The next time you re-open it the service connection will fail again.

I wanted to debug the error directly on my android phone, so I compiled a "debug" build (release builds cannot be debugged remotely) of v1.4.0 and installed it on my device. However, I'm unable to replicate the error using the debug build, which means I cannot debug the error to find out why it might be occurring! The code is exactly the same as the release build that was failing... 😖 

So then I decided to test a few different SSL configurations using api-beta.xbrowsersync.org (running v1.1.5), and using v1.4.0 of the app (release build), here are the results:

1. API running over https:443 - NO ISSUES
2. API running over https:50773 - ISSUE REPLICATED (initially, but then seems to have resolved itself)
3. API running over http:8080 behind a reverse proxy exposed as https:443 - NO ISSUES
4. API running over http:8080 behind a reverse proxy exposed as https:50773 - NO ISSUES

What's even weirder is that the problem is intermittent, for me when running scenario 2, the problem does not occur initially, it occurred after a few hours when I opened the app and the service status showed as `error`. However, some time after that the connection was restored and has been working fine since! Note, during the whole time the service was always reachable via Chrome, and the browser extension never had a connection error occur.

In conclusion, I'm unable to draw any conclusion as to why this problem is occurring. I can only say that I do not believe the problem is with the xBrowserSync app or service, it is perhaps some edge case issue with the Android webview (which cordova apps run in), but after much googling I've been unable to find any information to back this hunch up. Without an error message to go on I can't get any further here, perhaps some more information will come to light in future that may help in resolving this.

In the meantime, I would suggest either running the your service over the standard https port (443) or even better, running behind a reverse proxy (running SSL from there rather than the xBrowserSync service) which you can then use expose on any port you wish (i.e. scenario 4 above). This is a much better configuration as web servers are far more secure than an exposed node service.

Even better, I'll be releasing a [docker](https://www.docker.com/) configuration for the API very soon which will automatically configure your service and a reverse proxy for you. Keep an eye on [this issue](https://github.com/xBrowserSync/API/issues/15) if you're interested.

**Comment by nero120** [comment]
Hi @cjnaz, give the new [xBrowserSync Docker image](https://hub.docker.com/r/xbrowsersync/api) a try. This will allow you to stand up a fully configured, production ready service (complete with api, database and reverse proxy) running over HTTPS. Hopefully that way you will no longer see these issues.

**Comment by cjnaz** [comment]
Thanks Rich. I'll give this a shot on my next API update.  For now,
everything is solid when running https on port 443.
